core/host_api/impl/child_storage_extension.cpp

```diff
@@ -0,0 +1,231 @@
+      const common::Buffer &child_storage_key, F func, Args &&... args) const {
+    OUTCOME_TRY(prefixed_child_key,
+                make_prefixed_child_storage_key(child_storage_key));
+    OUTCOME_TRY(child_batch,
+                storage_provider_->getChildBatchAt(prefixed_child_key));
+
+    auto result = func(child_batch, std::forward<Args>(args)...);
+    if (not result) {
+      storage_provider_->clearChildBatches();
+      return result;
+    }
+
+    OUTCOME_TRY(new_child_root, child_batch->commit());
+    OUTCOME_TRY(storage_provider_->getCurrentBatch()->put(
+        prefixed_child_key,
+        common::Buffer{scale::encode(new_child_root).value()}));
+    storage_provider_->clearChildBatches();
+    if constexpr (!std::is_void_v<R>) {
+      return result;
+    }
+    return outcome::success();
+  }
+
+  template <typename... Args>
+  auto ChildStorageExtension::toBuffer(runtime::Memory &memory,
+                                       Args &&... spans) const {
+    auto f = [&memory](auto &&span) {
+      auto [span_ptr, span_size] = runtime::PtrSize(span);
+      return memory.loadN(span_ptr, span_size);
+    };
+    return std::make_tuple(f(std::forward<Args>(spans))...);
+  }
+
+  void ChildStorageExtension::ext_default_child_storage_set_version_1(
+      runtime::WasmSpan child_storage_key,
+      runtime::WasmSpan key,
+      runtime::WasmSpan value) {
+    auto &memory = memory_provider_->getCurrentMemory()->get();
+    auto [child_key_buffer, key_buffer, value_buffer] =
+        toBuffer(memory, child_storage_key, key, value);
+
+    SL_TRACE_VOID_FUNC_CALL(
+        logger_, child_key_buffer, key_buffer, value_buffer);
+
+    auto result = executeOnChildStorage<void>(
+        child_key_buffer,
+        [](auto &child_batch, auto &key, auto &value) {
+          return child_batch->put(key, value);
+        },
+        key_buffer,
+        value_buffer);
+
+    if (not result) {
+      logger_->error(
+          "ext_default_child_storage_set_version_1 failed with reason: {}",
+          result.error().message());
+    }
+  }
+
+  runtime::WasmSpan
+  ChildStorageExtension::ext_default_child_storage_get_version_1(
+      runtime::WasmSpan child_storage_key, runtime::WasmSpan key) const {
+    auto &memory = memory_provider_->getCurrentMemory()->get();
+    auto [child_key_buffer, key_buffer] =
+        toBuffer(memory, child_storage_key, key);
+
+    SL_TRACE_VOID_FUNC_CALL(logger_, child_key_buffer, key_buffer);
+
+    auto result = executeOnChildStorage<common::Buffer>(
+        child_key_buffer,
+        [](auto &child_batch, auto &key) -> outcome::result<common::Buffer> {
+          return child_batch->get(key);
+        },
+        key_buffer);
+    auto option = result ? std::make_optional(result.value()) : std::nullopt;
+    if (result) {
+      SL_TRACE_FUNC_CALL(logger_, option.value(), child_key_buffer, key_buffer);
+    } else {
+      logger_->trace(
```

Comment:
It's a copy from StorageExtension::ext_storage_get_version_1. So I'm not sure. Is it even something worth to be warned about?

Reply:
I think there's a mistake in storage_get: we should only do so If the error is NOT_FOUND, but for other errors we better log them properly and perhaps throw and exception to terminate the current runtime call, since what sense would it make If the call failed.

Reply:
I looked into internals a bit.
The only error we can possibly get here is TrieError::NO_VALUE. No others. So, that's what we trace and I think it makes sense.
The only other result we might get is unexpected exception.

Reply:
The problem is there is no explicit contract in the code that we can get only TrieError::NO_VALUE, so it's better to explicitly filter it here.

Reply:
Updated.